src/librustdoc/passes/calculate_doc_coverage.rs

```diff
@@ -245,7 +257,12 @@ impl fold::DocFolder for CoverageCalculator {
                     has_docs,
                     has_doc_example,
                     should_have_doc_example(&i.inner),
+                    should_have_docs,
                 );
+
+                if !should_have_docs {
+                    return Some(i);
+                }
```

Comment:
Hmm, I know I suggested this solution, but I'm not sure it's actually correct ... This will no longer count nested items with documentation. Maybe instead this should store a `allow_missing` state?

cc @guillaumegomez, I think this is missing for `allow_missing_examples` too.

Reply:
I was wondering: shouldn't it run `self.fold_item_recur(i)` instead? (maybe it's what you said?)

Reply:
This has two conflicting goals:
1. For nested items missing docs, it should ignore the missing docs:
```rust
#[allow(missing_docs)]
impl MyStruct {
    fn f() {}
}
```
2. For nested items that _have_ docs, it should count those, even if the parent item allowed docs to be missing:
```
#[allow(missing_docs)]
impl MyStruct {
     /// This is documented
    fn f() {}
}
```
Currently the code does 1. but not 2. If it just ran `fold_item_recur` it would do 2. but not 1. I'm suggesting a way to do both.

Reply:
Oh I see, we have two different visions here. For me `#[allow(missing_docs)]` only applies to the item, not its children.

Reply:
See https://github.com/rust-lang/rust/pull/76196#issuecomment-684884304

Reply:
This is also not taking into account things like `#[warn(missing_docs)]` on inner items which should be counted in the coverage. We might be better off using something like [`lint_level_at_node`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TyCtxt.html#method.lint_level_at_node) to find out if `missing_docs` is allowed on any particular item (making sure [`LintSource`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/lint/enum.LintSource.html) is `Node`). Then this can continue to recurse as normal.

Reply:
@r-52 can you change this to use `lint_level_at_node` so this is consistent with the rest of the `rustc` lints?

Separately, rustdoc should probably also use this for `allow(missing_doc_code_examples)`, it makes me slightly uncomfortable those are separate passes.

Reply:
Ok actually I misread, they're both in the same pass here. Do you mind fixing the problem for code examples too? No problem if not, I can fix it in a follow-up.